li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -771,4 +818,40 @@ private void ensureOpen() {
       throw new IllegalStateException("this federated consumer has already been closed");
     }
   }
+
+  // TODO: check the client registration state and handle accordingly if it is the fallback mode
+  private void maybeResubscribe() {
```

Comment:
names that start with maybe ..... can we call this "refreshSubscriptionMetadata" ? (or assignment metadata, or something along those lines)

Reply:
sure. done.